scout/server/utils.py

```diff
@@ -134,24 +134,31 @@ def case_append_bam(case_obj):
         for bam in bam_files:
             bam_path = individual.get(bam[0])
             if not (bam_path and os.path.exists(bam_path)):
-                LOG.debug("%s: no bam file found", individual['individual_id'])
+                LOG.debug("%s: no bam/cram file found", individual['individual_id'])
                 continue
             case_obj[bam[1]].append(bam_path) # either bam_files or mt_bams
-            case_obj[bam[2]].append(find_bai_file(bam_path)) # either bai_files or mt_bais
+            case_obj[bam[2]].append(find_index(bam_path)) # either bai_files or mt_bais
 
-def find_bai_file(bam_file):
+
+def find_index(align_file):
     """Find out BAI file by extension given the BAM file.
 
-    Index files wither ends with filename.bam.bai or filename.bai
+    Index files wither ends with filename.bam.bai or filename.bai /
+    In case of cram alignments the index is named filename.cram.crai or filename.crai
 
     Args:
-        bam_file(str): The path to a bam file
+        align_file(str): The path to a bam/cram file
 
     Returns:
-        bai_file(str): Path to index file
+        index_file(str): Path to index file
     """
-    bai_file = bam_file.replace('.bam', '.bai')
-    if not os.path.exists(bai_file):
-        # try the other convention
-        bai_file = "{}.bai".format(bam_file)
-    return bai_file
+    index_file = None
+    if align_file.endswith('cram'):
+        index_file = align_file.replace('.cram', '.crai')
+        if not os.path.exists(index_file):
+            index_file = "{}.crai".format(align_file)
+    else:
+        index_file = align_file.replace('.bam', '.bai')
+        if not os.path.exists(index_file):
+            index_file = "{}.bai".format(align_file)
```

Comment:
..and that one never gets tested for existence. I know its the same as before really, but perhaps a good time to fix that and warning/error handling?

Reply:
I'll try to handle it, thanks!

Reply:
But I'm not flashing messages from here, because this is called in a loop, for instance from the str variants page. I'll handle it from the view instead

Reply:
@dnil see if you like this way of handling missing files, otherwise I'm open to suggestions!

Reply:
Ehr, am I reading this backwards, or isn't this is basically the same. There is a test in the view now, but index_file will always be set regardless `if os.path.exists(that_fallback_index_path), so the test will still pass even if an index here or there is missing?

Reply:
The loophole is still there. Its not super important, but...

Reply:
Well, I give in. It is working, I don't really think we promise to test user given paths anyway, and the test that is in place is better than none. As you say, worst case appears to be that igv.js fails/reports error.

Reply:
Thanks for the review!